gclient/src/api/listener/mod.rs

```diff
@@ -58,30 +62,64 @@ impl From<GenDispatchStatus> for DispatchStatus {
 }
 
 impl DispatchStatus {
+    /// Check whether `DispatchStatus` is
+    /// [`Success`](DispatchStatus::Success)ful.
     pub fn succeed(&self) -> bool {
         matches!(self, DispatchStatus::Success)
     }
 
+    /// Check whether `DispatchStatus` is [`Failed`](DispatchStatus::Failed).
     pub fn failed(&self) -> bool {
         matches!(self, DispatchStatus::Failed)
     }
 
+    /// Check whether `DispatchStatus` is
+    /// [`NotExecuted`](DispatchStatus::NotExecuted).
     pub fn not_executed(&self) -> bool {
         matches!(self, DispatchStatus::NotExecuted)
     }
 }
 
+/// Events processing trait.
+///
+/// This trait provides default implementations for processing several events
```

Comment:
Not especially for networks. It may be mocked etc. Just processing some iterator of the events

Reply:
Fixed